core/src/main/java/org/bitcoinj/wallet/Wallet.java

```diff
@@ -2182,39 +2185,28 @@ public void notifyNewBestBlock(StoredBlock block) throws VerificationException {
         lock.lock();
         try {
             // Store the new block hash.
+            context.setBlockChainHeight(block.getHeight());
             setLastBlockSeenHash(newBlockHash);
             setLastBlockSeenHeight(block.getHeight());
             setLastBlockSeenTimeSecs(block.getHeader().getTimeSeconds());
             // Notify all the BUILDING transactions of the new block.
             // This is so that they can update their depth.
-            Set<Transaction> transactions = getTransactions(true);
-            for (Transaction tx : transactions) {
+            for (Transaction tx : this.transactions.values()) {
                 if (ignoreNextNewBlock.contains(tx.getHash())) {
                     // tx was already processed in receive() due to it appearing in this block, so we don't want to
                     // increment the tx confidence depth twice, it'd result in miscounting.
                     ignoreNextNewBlock.remove(tx.getHash());
                 } else {
-                    TransactionConfidence confidence = tx.getConfidence();
-                    if (confidence.getConfidenceType() == ConfidenceType.BUILDING) {
-                        // Erase the set of seen peers once the tx is so deep that it seems unlikely to ever go
-                        // pending again. We could clear this data the moment a tx is seen in the block chain, but
-                        // in cases where the chain re-orgs, this would mean that wallets would perceive a newly
-                        // pending tx has zero confidence at all, which would not be right: we expect it to be
-                        // included once again. We could have a separate was-in-chain-and-now-isn't confidence type
-                        // but this way is backwards compatible with existing software, and the new state probably
-                        // wouldn't mean anything different to just remembering peers anyway.
-                        if (confidence.incrementDepthInBlocks() > context.getEventHorizon())
-                            confidence.clearBroadcastBy();
-                        confidenceChanged.put(tx, TransactionConfidence.Listener.ChangeReason.DEPTH);
-                    }
+                    confidenceChanged.put(tx, TransactionConfidence.Listener.ChangeReason.DEPTH);
                 }
             }
 
             informConfidenceListenersIfNotReorganizing();
             maybeQueueOnWalletChanged();
 
             if (hardSaveOnNextBlock) {
-                saveNow();
+                //saveNow();
```

Comment:
hardSaveOnNextBlock gets completely overridden here. at least the if is pointless.

Reply:
That was not branch for PR... Some more work supposed to be :)
I suppose we should discuss about this saving...